src/collections/componentSet.ts

```diff
@@ -662,6 +671,52 @@ export class ComponentSet extends LazyCollection<MetadataComponent> {
     }
     return destructiveChangesTypes;
   }
+
+  /**
+   * Returns an API version to use as the value of the `version` field
+   * in a manifest (package.xml) for MDAPI calls in the following order
+   * of preference:
+   *
+   * 1. this.sourceApiVersion
+   * 2. this.apiVersion
+   * 3. sourceApiVersion set in sfdx-project.json
+   * 4. apiVersion from ConfigAggregator (config files and Env Vars)
+   * 5. http call to apexrest endpoint for highest apiVersion
+   * 6. hardcoded value of "58.0" as a last resort
+   *
+   * @returns string The resolved API version to use in a manifest
+   */
+  private async getApiVersion(): Promise<string> {
+    let version = this.sourceApiVersion ?? this.apiVersion;
+
+    if (!version) {
+      try {
+        const project = await SfProject.resolve();
+        const projectConfig = await project.resolveProjectConfig();
+        version = projectConfig?.sourceApiVersion as string;
+      } catch (e) {
+        // If there's any problem just move on to ConfigAggregator
+      }
+    }
+
+    if (!version) {
+      try {
+        version = ConfigAggregator.getValue(OrgConfigProperties.ORG_API_VERSION).value as string;
```

Comment:
I think you can say `ConfigAggregator.getValue<string>`

Reply:
apparently that's `getPropertyValue()`